production.py

```diff
@@ -0,0 +1,27 @@
+# The COPYRIGHT file at the top level of this repository contains the full
+# copyright notices and license terms.
+from trytond.pool import PoolMeta
+from trytond.model import fields
+from .stock_lot import CreateQualityLotTestsMixin
+
+
+class Template(metaclass=PoolMeta):
+    __name__ = 'product.template'
+    production_quality_template = fields.Many2One('quality.template',
+        'Production Quality Template')
+
+
+class Production(CreateQualityLotTestsMixin, metaclass=PoolMeta):
+    __name__ = 'production'
+
+    @classmethod
+    def done(cls, productions):
+        super().done(productions)
+        cls.create_lot_quality_tests(productions, 'production')
+
+    def lots_for_quality_tests(self):
+        return list(set(m.lot for m in self.outputs if m.lot and
+            m.state == 'done' and
+            m.product.template.production_quality_template and
+            not [x for x in m.lot.quality_tests if
+            m.product.template.production_quality_template in x]))
```

Comment:
Are you sure we should check that the quality template is the same? Isn't it enough to check that a quality test exists?

Reply:
It is true that in 99 percent of cases it is enough to see if there is a test, but in some cases it may be possible to pass quality tests on entry and production, just re-label the product 

We can leave by default the verification that only tests exist, but then only allow in the product to choose one.